lib/fog/aws/credential_fetcher.rb

```diff
@@ -58,6 +56,23 @@ def fetch_credentials(options)
             super
           end
         end
+
+        def fetch_credentials_token_header(connection, disable_imds_v2)
+          return nil if disable_imds_v2
+
+          token = connection.put(
+            :path => INSTANCE_METADATA_TOKEN,
+            :idempotent => true,
+            :expects => 200,
+            :retry_interval => 1,
+            :retry_limit => 3,
```

Comment:
Should you add a `read_timeout` here? Excon has a default read time out of 60 seconds and this is too long a time to wait to verify if the first put option works.

Reply:
I've added more timeouts now